src/main/java/uk/gov/pay/api/resources/HalResourceBuilder.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.pay.api.resources;
+
+import black.door.hate.HalRepresentation;
+import black.door.hate.HalRepresentation.HalRepresentationBuilder;
+import black.door.hate.HalResource;
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class HalResourceBuilder implements HalResource {
+    public static final String SELF_LINK_KEY = "self";
+    private Map<String, URI> linkMap = new HashMap<>();
+    private Map<String, Object> propertyMap = new HashMap<>();
+    private URI selfLink;
+
+    public HalResourceBuilder(URI selfLink) {
+        try {
+            this.selfLink = selfLink != null ? selfLink: new URI("");
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public HalResourceBuilder withLink(String linkKey, URI link) {
+        if (linkKey!=null && link !=null) {
+            linkMap.put(linkKey, link);
+        }
+        return this;
+    }
+
+    public HalResourceBuilder withProperty(String key, Object value) {
+        if (key!=null && value!=null) {
+            propertyMap.put(key, value);
+        }
+        return this;
+    }
+
+    @Override
+    public URI location() {
+        return selfLink;
+    }
+
+    @Override
+    public HalRepresentationBuilder representationBuilder() {
+        HalRepresentationBuilder builder = HalRepresentation.builder();
+        builder.addLink(SELF_LINK_KEY, this);
+
+        for (String key : linkMap.keySet()) {
+            builder.addLink(key, linkMap.get(key));
+        }
+        for (String key : propertyMap.keySet()) {
+            builder.addProperty(key, propertyMap.get(key));
+        }
+        return builder;
+    }
+
+    public String build() {
+        try {
+            return representationBuilder().build().serialize();
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
```

Comment:
Log the exception and pass the exception into it rather than doing a printStackTrace